core/DataAccess/ArchiveSelector.php

```diff
@@ -205,10 +205,14 @@ public static function getArchiveIds($siteIds, $periods, $segment, $plugins, $in
                 $archives = $row['archives'];
                 $pairs = explode(',', $archives);
                 foreach ($pairs as $pair) {
-                    list($idarchive, $doneFlag) = explode('|', $pair);
+                    list($idarchive, $doneFlag, $value) = explode('|', $pair);
 
                     $result[$doneFlag][$dateStr][] = $idarchive;
-                    if (strpos($doneFlag, '.') === false) { // all plugins archive
+                    if (strpos($doneFlag, '.') === false // all plugins archive
+                        // sanity check: DONE_PARTIAL shouldn't be used w/ done archives, but in case we see one,
+                        // don't treat it like an all plugins archive
+                        && $value != ArchiveWriter::DONE_PARTIAL
+                    ) {
```

Comment:
This and the above code is a safety check for if an archive has a `done` like flag but `DONE_PARTIAL` value. In case of problematic data like this, we just treat it like another partial archive.